File: lang_py/Harddrive.py
from Komponen import Komponen

class Harddrive(Komponen):

    def __init__(self, merk, nama, kapasitasGB, tipeDrive):
        super().__init__(merk, nama)
        self.kapasitasGB = kapasitasGB
        self.tipeDrive = tipeDrive

    def setKapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB

    def getKapasitasGB(self):
        return self.kapasitasGB
    
    def setTipeDrive(self, tipeDrive):
        self.tipeDrive = tipeDrive
    
    def getTipeDrive(self):
        return self.tipeDrive
    

File: lang_java/Harddrive.java
class Harddrive extends Komponen{

    private int kapasitasGB;
    private String tipeDrive;

    public Harddrive(){
    }

    public Harddrive(String merk, String nama, int kapasitasGB, String tipeDrive){
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.tipeDrive = tipeDrive;
    }

    public void setKapasitasGB(int kapasitasGB){
        this.kapasitasGB = kapasitasGB;
    }

    public int getKapasitasGB(){
        return kapasitasGB;
    }

    public void setTipeDrive(String tipeDrive){
        this.tipeDrive = tipeDrive;
    }

    public String getTipeDrive(){
        return tipeDrive;
    }

}